bool clientConnection
bool thread
bool bufferSent
